### PR TITLE
FR-16888 - Secure Token Exchange via Middleware

### PR DESCRIPTION
**Pull request by frontegg-david**

### CODE REVIEW

File: packages/nextjs/src/api/index.ts [hunk 1]
```diff
@@ -41,12 +41,42 @@ const refreshTokenEmbedded = async (headers: Record<string, string>) => {
...
+  refresh_token: string,
+  cliendId?: string,
+  clientSecret?: string
+) => {
   return Post({
     url: `${config.baseUrl}${ApiUrls.refreshToken.hosted}`,
     body: JSON.stringify({
       grant_type: 'refresh_token',
       refresh_token,
+      client_id: cliendId,
```
[the remark this answers is not in the document]
Comment (frontegg-david) [reply packages/nextjs/src/api/index.ts:55]: I used it with clientId and secret only if secureJwtEnabled

[the remark this answers is not in the document]
Comment (frontegg-david) [reply packages/nextjs/src/api/index.ts:55]:
```
if (config.secureJwtEnabled) {
      const clientId = config.clientId;
      const clientSecret = config.clientSecret;

      logger.info('going to refresh token (hosted-login mode) (secure-jwt mode)');
      return await api.refreshTokenHostedLogin(headers, tokens.refreshToken, clientId, clientSecret);
    } else {
      logger.info('going to refresh token (hosted-login mode) ', tokens.refreshToken);
      return await api.refreshTokenHostedLogin(headers, tokens.refreshToken);
    }
```

File: packages/nextjs/src/middleware/helpers.ts [hunk 2]
```diff
@@ -57,3 +57,65 @@ export const getHostedLogoutUrl = (referer = config.appUrl): BuildRouteResult =>
...
+  if (body.authResponse) {
+    Object.assign(body, body.authResponse);
+  }
+  return {
+    accessToken: body.accessToken || body.access_token,
+    refreshToken: body.refreshToken || body.refresh_token,
+  };
+};
+
+const jwtKeys = ['accessToken', 'access_token', 'idToken', 'id_token'];
```
Comment (aviadmizrachi) [inline packages/nextjs/src/middleware/helpers.ts:82]: Just note that in some cases the accessToken is not a JWT

**Review by aviadmizrachi — APPROVED** [review]

File: packages/nextjs/src/middleware/helpers.ts (hunk 2 above)
Comment (frontegg-david) [reply packages/nextjs/src/middleware/helpers.ts:82]: access token in implicit mode is still a JWT

**Pull request merged**